airbyte-workers/src/main/java/io/airbyte/workers/internal/EmptyAirbyteSource.java

```diff
@@ -5,29 +5,81 @@
 package io.airbyte.workers.internal;
 
 import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.ResetSourceConfiguration;
+import io.airbyte.config.StateType;
+import io.airbyte.config.StateWrapper;
+import io.airbyte.config.StreamDescriptor;
 import io.airbyte.config.WorkerSourceConfig;
+import io.airbyte.config.helpers.StateMessageHelper;
+import io.airbyte.protocol.models.AirbyteGlobalState;
 import io.airbyte.protocol.models.AirbyteMessage;
 import io.airbyte.protocol.models.AirbyteMessage.Type;
 import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.AirbyteStreamState;
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedList;
 import java.util.Optional;
+import java.util.Queue;
+import java.util.Set;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
 
 /**
  * This source will never emit any messages. It can be used in cases where that is helpful (hint:
  * reset connection jobs).
  */
+@Slf4j
 public class EmptyAirbyteSource implements AirbyteSource {
 
   private final AtomicBoolean hasEmittedState;
+  private final Queue<StreamDescriptor> streamsToReset = new LinkedList<>();
+  // TODO: Once we are sure that the legacy way of transmitting the state is not use anymore, we need
+  // to remove this variable and the associated
+  // checks
+  private boolean isResetBasedForConfig;
```

Comment:
i would still like to see a comment explaining what this means, not just the todo about the condition under which it will be removed. while it is in the codebase a developer needs to be able to understand what it is and right now it is not clear.

suggestion
```
  /**
   * In the legacy version the config was empty. The new one has a populated config. This variable
   * encapsulates whether we are in the legacy version or not.
   */
```

Reply:
Added